### DISCUSSION TITLE
How can I get a translation outside of a component in NextJS13

### DISCUSSION THREAD

**Discussion by pepew-le-boss** [vinissimus/next-translate#1089] [General] [answered]
I've came across  the `getT()` function but I don't get how to use it since it's a promise.

Here's an example:

```
import getT from "next-translate/getT"

const t = await getT("homepage")

export const metadata: Metadata = {
  title: t("main_title"),
}

export default async function HomePage() {
...
}
```

I get the following error: `Top-level 'await' expressions are only allowed`

My question is do I need to use another function or do I need to configure my project so I can use top-level awaits ?

**Comment by aralroca** [comment, marked as answer]
Instead of metadata you can use :

```js
export async function generateMetadata({ params }) {
  const t = await getT(params.lang, "homepage")
  return {
    title:t("main_title")
  }
}
```
